EcommerceStore.Application/Services/BrandService.cs

```diff
@@ -0,0 +1,110 @@
+using EcommerceStore.Application.Exceptions;
+using EcommerceStore.Application.Interfaces;
+using EcommerceStore.Application.Models.InputModels;
+using EcommerceStore.Application.Models.ViewModels;
+using EcommerceStore.Domain.Entities;
+using EcommerceStore.Domain.Interfaces;
+
+namespace EcommerceStore.Infrastucture.Services
+{
+    public class BrandService : IBrandService
+    {
+        private readonly IBrandRepository _brandRepository;
+
+        public BrandService(IBrandRepository brandRepository)
+        {
+            _brandRepository = brandRepository;
+        }
+
+        public async Task CreateBrandAsync(BrandInputModel brandIm)
+        {
+            var brand = new Brand
+            {
+                Name = brandIm.Name,
+                FoundationYear = brandIm.FoundationYear
+            };
+
+            var brands = await _brandRepository.GetAllAsync();
```

Comment:
it is better to create method getBrandByBrandName rather than retrieving all the brands from db